app/models/user.rb

```diff
@@ -74,6 +74,7 @@ def school
   scope :for_lead_provider, -> { includes(:lead_provider).joins(:lead_provider) }
   scope :admins, -> { joins(:admin_profile) }
   scope :early_career_teachers, -> { joins(:early_career_teacher_profile).includes(:early_career_teacher_profile) }
+  scope :mentors, -> { joins(:mentor_profile).includes(:mentor_profile) }
```

Comment:
As mentioned earlier - this type of scope is only required because we query User in the controller instead of profile. I'd see a User as a record used for authentication (and email), but apart of that it should be pretty much logic-less.